messaging/src/test/java/org/axonframework/eventhandling/scheduling/jobrunr/JobRunrEventSchedulerTest.java

```diff
@@ -0,0 +1,191 @@
+               .useJobActivator(new SimpleActivator(eventScheduler))
+               .useStorageProvider(storageProvider)
+               .useBackgroundJobServer(usingStandardBackgroundJobServerConfiguration().andPollIntervalInSeconds(5))
+               .initialize();
+        backgroundJobServer = JobRunr.getBackgroundJobServer();
+    }
+
+    @Test
+    void whenScheduleIsCalledThanShouldPublishEvent() {
+        eventScheduler.schedule(Duration.ZERO, 1);
+        Instant rightAfterSchedule = Instant.now();
+
+        await().atMost(Duration.ofSeconds(2)).until(() -> !publishedMessages.isEmpty());
+        assertEquals(1, publishedMessages.size());
+
+        EventMessage<?> publishedMessage = publishedMessages.get(0);
+
+        assertEquals(1, publishedMessage.getPayload());
+        assertTrue(rightAfterSchedule.isBefore(publishedMessage.getTimestamp()));
+        assertTrue(publishedMessage.getMetaData().isEmpty());
+    }
+
+    @Test
+    void whenScheduleIsCalledWithEventMessageMetadataShouldBePreserved() {
+        Map<String, Object> metadata = new HashMap<>();
+        metadata.put("foo", "bar");
+        EventMessage<?> originalMessage = new GenericEventMessage<>(2, metadata);
+        eventScheduler.schedule(Instant.now(), originalMessage);
+        Instant rightAfterSchedule = Instant.now();
+
+        await().atMost(Duration.ofSeconds(2)).until(() -> !publishedMessages.isEmpty());
+        assertEquals(1, publishedMessages.size());
+
+        EventMessage<?> publishedMessage = publishedMessages.get(0);
+
+        assertEquals(2, publishedMessage.getPayload());
+        assertTrue(rightAfterSchedule.isBefore(publishedMessage.getTimestamp()));
+        assertEquals(metadata, publishedMessage.getMetaData());
+    }
+
+    @Test
+    void rescheduleWithDurationShouldWork() {
+        ScheduleToken token = eventScheduler.schedule(Duration.ofMillis(100L), 3);
+        eventScheduler.reschedule(token, Duration.ofMillis(100L), 4);
+
+        await().atMost(Duration.ofSeconds(2)).until(() -> !publishedMessages.isEmpty());
+        assertEquals(1, publishedMessages.size());
+
+        EventMessage<?> publishedMessage = publishedMessages.get(0);
+        assertEquals(4, publishedMessage.getPayload());
+    }
+
+    @Test
+    void rescheduleWithInstantShouldWork() {
+        ScheduleToken token = eventScheduler.schedule(Instant.now().plusMillis(100L), 5);
+        eventScheduler.reschedule(token, Instant.now().plusMillis(100L), 6);
+
+        await().atMost(Duration.ofSeconds(2)).until(() -> !publishedMessages.isEmpty());
+        assertEquals(1, publishedMessages.size());
+
+        EventMessage<?> publishedMessage = publishedMessages.get(0);
+        assertEquals(6, publishedMessage.getPayload());
+    }
+
+    @Test
+    void incorrectTokenClassShouldThrow() {
+        ScheduleToken token = new SimpleScheduleToken("ff");
+        assertThrows(IllegalArgumentException.class, () -> eventScheduler.cancelSchedule(token));
+    }
+
```

Comment:
Nit: can you add a test for the `shutdown()` method too?

Reply:
Added